RepoDb.Core/RepoDb/Extensions/DbCommandExtension.cs

```diff
@@ -372,7 +377,11 @@ public static class DbCommandExtension
             var typeHandler = PropertyHandlerCache.Get<object>(valueType);
             if (typeHandler != null)
             {
-                // TODO: Ensure to reuse the existing PropertyHandler (if given)
+                // It is hard to pre-compile this as the property handler is dynamic
+                var setMethod = typeHandler.GetType().GetMethod("Set");
+                var returnType = setMethod.ReturnType;
+                dbType = m_clientTypeToDbTypeResolver.Resolve(returnType);
+                value = setMethod.Invoke(typeHandler, new[] { value, null });
```

Comment:
Is the `null` right? @mikependon

Reply:
It is important to pass the instance of ClassProperty as part of the context of property handler execution.

Do you have the instance of ProperyInfo or even just a name of the property? If yes, then please use the PropertyCache.Get(entityType) to get the list of ClassProperty objects. From there, you can find the equivalent ClassProperty object by iterating the list of properties and by equating the property mapped name to ClassProperty.GetMappedName() method of each instance.

Reply:
FYI: Use the PropertyMappedNameCache.Get(PropertyInfo) to get the mapped name of the PropertyInfo object.

Reply:
@mikependon Thank you for your response. Over which properties should it iterate, there is no context or relation to the type.

Reply:
Hmmm, upon realizing here, I think I had replied to you wrongly. Let me debug this one now. :)

Reply:
What you had made here is correct. For the `QueryField`, `IEnumerable<QueryField>`, `QueryGroup` and `Dictionary<string, object>` parameter that does not have the property property entity type mapping, it should be ignoring the `ClassProperty`. I would humbly say that only the type level must be handled - specifically on this case.